### PR TITLE
Add a Robot Detector 

### PR DESCRIPTION
**Pull request by ysims**
This PR adds a detector that uses the Visual Mesh to find the position of robots in the scene. 

Basic detector that clusters the points, does a few throwouts (hull, distance, number of points in the cluster) and then finds the closest point to use as the position of the robot. 

Added to NUsight, just a cone/circle to visualise the position similarly to the balls. 

### Pre-PR Checklist:

Have you

- [ ] Updated NUbook if necessary (add link to NUbook PR here) 
- [x] Added/updated tests for your changes, including regression tests for bug fixes
- [x] Updated relevant module READMEs
- [x] Added/modified [documentation directives](https://nubook.nubots.net/guides/general/code-conventions#documentation) in relevant code
- [x] Added a descriptive title and relevant labels to the PR

### Future Enhancements

- Better NUsight visualisation. Proper radius, and maybe visualise it more as a circle on the ground? Like video game character switching often has a highlighted circle. Colour of the circle matching team colour? And adding this to the localisation view.
- Find robot height and width and utilise these, particularly to get centre position of robot and for throwouts
- Dealing with robots where the cluster has separated due to the legs being thin
- Filtering/localisation of robots
- Utilise robot positions in behaviour

### CODE REVIEW

**Comment by jpptm** [comment]
got any sample outputs that we can have a look at?

**Comment by ysims** [comment]
@jpptm 

image

image

image

File: module/vision/RobotDetector/src/RobotDetector.cpp [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                if (robot.rRCc.norm() < cfg.minimum_robot_distance) {
+                    log<NUClear::DEBUG>(fmt::format("Cluster rejected due to distance: {}. Min allowed distance {}.",
+                                                    robot.rRCc.norm(),
+                                                    cfg.minimum_robot_distance));
+                    continue;
+                }
+
+                // Should be changed to the average confidence of the robot
+                robot.covariance = Eigen::Matrix3d::Ones();
+                robot.radius     = 0.05;
```
Comment (jpptm) [inline module/vision/RobotDetector/src/RobotDetector.cpp:145]: Should this be a configurable value? Otherwise LGTM

**Review by jpptm — APPROVED** [review]
lgtm

**Pull request merged**